src/components/PaymentCredit/PaymentCredit.tsx

```diff
@@ -0,0 +1,182 @@
+import React from 'react'
+import './PaymentCredit.scss'
+import {Form, Field} from 'react-final-form'
+import formatString from "format-string-by-pattern";
+
+interface IProps {
+    setCardCredit: any;
+    submitBtn: any;
+    validate: any;
+    setCard: any;
+    cardCredit: string;
+    cardValue: string;
+}
+
+const PaymentCredit:React.FC<IProps> = ({validate, submitBtn, setCard, setCardCredit, cardValue}) => {
+
+
+    const clearNumber = (value = '') => {
+        return value.replace(/\D+/g, '')
+    }
+
+    const cardValidate = (input:any) => {
+
+        if(cardValue === "Visa card"){
+            let cardno =  /^(?:4[0-9]{12}(?:[0-9]{3})?)$/
```

Comment:
You should move this to constants and name them something like "visaValidator", "masterValidator" etc